### PR TITLE
Add work api spec and client

### PR DESCRIPTION
**Pull request by qiujian16**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+# Build docker containers
+.PHONY: docker-build
+docker-build: generate fmt vet manifests
+	docker build . -t ${IMG}
+
+# Push the docker image
+.PHONY: docker-push
+docker-push: docker-build
+	docker push ${IMG}
```
Comment (RainbowMango) [inline Makefile:71]: Just a reminder, the `Dockfile` is missed.

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,9 @@
+module sigs.k8s.io/work-api
+
+go 1.15
+
+require (
+	k8s.io/apimachinery v0.20.4
+	k8s.io/client-go v0.20.2
```
Comment (RainbowMango) [inline go.mod:7]:
```suggestion
	k8s.io/apimachinery v0.20.4
	k8s.io/client-go v0.20.4
```

Comment (RainbowMango) [inline go.mod:7]:
```suggestion
	k8s.io/apimachinery v0.20.4
	k8s.io/client-go v0.20.4
```

File: hack/boilerplate/boilerplate.py [hunk 3]
```diff
@@ -0,0 +1,215 @@
+#!/usr/bin/env python
+
+# Copyright 2015 The Kubernetes Authors.
```
Comment (RainbowMango) [inline hack/boilerplate/boilerplate.py:3]: Copyright 2015?

**Review by RainbowMango — COMMENTED** [review]
Great Job! @qiujian16 

Just some nits.  Copyright year should be consistent, and `2021` with newly created files?

File: hack/boilerplate/boilerplate.py (hunk 3 above)
Comment (qiujian16) [reply hack/boilerplate/boilerplate.py]: fixed

File: Makefile (hunk 1 above)
Comment (qiujian16) [reply Makefile:71]: thanks for reminding, i will add that when we have a binary built in the next PR.

**Comment by RainbowMango** [comment]
/lgtm
Thanks~
@pmorie Could you please take a look?

**Pull request merged**